sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StateStoreSuite.scala

```diff
@@ -1718,6 +1845,240 @@ abstract class StateStoreSuiteBase[ProviderClass <: StateStoreProvider]
     assert(encoderSpec == deserializedEncoderSpec)
   }
 
+  test("SPARK-51596: task thread waits for ongoing maintenance to complete before" +
```

Comment:
rename this, the task thread isn't the one waiting. Task thread just submits to maint threadpool